lib/pure/strutils.nim

```diff
@@ -2298,6 +2298,54 @@ proc removeSuffix*(s: var string, suffix: string) {.
     newLen -= len(suffix)
     s.setLen(newLen)
 
+proc removePrefix*(s: var string, chars: set[char] = Newlines) {.
+  rtl, extern: "nsuRemovePrefixCharSet".} =
+  ## Removes the first matching character from the start of a string (in-place)
+  ## given a set of characters. If the set of characters is only equal to
+  ## `Newlines` then it will remove both the newline and return feed.
+  ## .. code-block:: nim
+  ##   var
+  ##     userInput = "\r\n*Hello World!"
+  ##     otherInput = "?!?Hello!?!"
+  ##   userInput.removePrefix
+  ##   userInput == "*Hello World!"
+  ##   userInput.removePrefix({'H', '*'})
+  ##   userInput == "Hello World!"
+  ##   otherInput.removePrefix({'!', '?'})
+  ##   otherInput == "!?Hello!?!"
+  if s.len == 0: return
+  var start = 0
+  if chars == Newlines:
+    if s[start] == '\13':
+      start += 1
+    if s[start] == '\10':
+      start += 1
+  else:
+    if s[start] in chars:
+      start += 1
+  s = s[start..s.high]
```

Comment:
Using `strutils.delete` here is more likely to avoid unnecessary allocation.

Reply:
Good call, I'll update